src/events/provider.py

```diff
@@ -122,7 +129,8 @@ def _on_relation_broken(self, event: RelationBrokenEvent) -> None:
         if not self.charm.unit.is_leader():
             return
 
-        auth = AuthenticationManager(self.context.auth_db)
+        # FIXME: There is not guarantee here
```

Comment:
WDYM? This relation (provider) relation is different from Kyuubi <> Postgres(Auth DB) relation -- and we block Kyuubi if it is not related over auth db interface anway.

Reply:
Kyuubi might be blocked from a status perspective, but there is nothing preventing the following scenario:
- We have a working setup with auth db and a client
- We remove the auth db relation, kyuubi goes into blocked status
- We remove the client relation. There is nothing preventing this event handler from running, so `self.context.auth_db` is None actually

I use `cast` as a way to tell the type checker "in this specific situation, trust me I know what type is my var", but here, it could be none.

Right now I care about getting the green tick, but subsequent PRs should focus on type exactitude. I should check whether `self.context.auth_db` is None or not, and then act on it

Reply:
I get your point. Thanks for the clarification.